File: 019_username/username.pl
#!/usr/bin/env perl

use common::sense;

while (<>) {

	chomp $_;
	my $len = length($_);

	my $is_letter = $_ =~ m/[A-Z]{1}/;
	my $is_digit  = $_ =~ m/\d{1}/;
	my $is_spec   = $_ =~ m/[@#*=]{1}/;
	my $is_space  = $_ =~ m/\s/;

	say (
		$is_letter && $is_digit && $is_spec && !$is_space && $len >= 5 && $len <= 10
			? 'PASS'
			: 'FAIL'
	);
}

File: 019_username/username.rb
#!/usr/bin/env ruby 

ARGF.each do |pass|

    pass.chomp!
	len       = pass.size 
	is_letter = pass.scan(/[A-Z]{1}/).size
	is_digits = pass.scan(/\d/).size
	is_spec   = pass.scan(/[@#*=]{1}/).size
	is_space  = pass.scan(/[ ]{1}/).size

	puts is_letter > 0 && is_digits > 0 && is_spec > 0 && is_space == 0 && len.between?(5,10) ? 'PASS' : 'FAIL'
end
